vernac/debugHook.ml

```diff
@@ -77,3 +77,72 @@ module Intf = struct
   let get () = !ltac_debug_ref
 
 end
+
+(* breakpoints as used by tactic_debug *)
+type breakpoint = {
+  dirpath : string;
+  offset : int;
+}
+
+let compare b1 b2 =
+  let c1 = Int.compare b1.offset b2.offset in
+  if c1 <> 0 then c1 else String.compare b1.dirpath b2.dirpath
+
+module BPSet = CSet.Make(struct
+  type t = breakpoint
+  let compare = compare
```

Comment:
Please inline the compare function here, the definition above is shadowing the default `compare` function.

Reply:
Like this?

```
module BPSet = CSet.Make(struct
  type t = breakpoint
  let compare =
    (fun b1 b2 ->
      let c1 = Int.compare b1.offset b2.offset in
      if c1 <> 0 then c1 else String.compare b1.dirpath b2.dirpath)
  end)
```

Reply:
Yes, up to the fact it's not idiomatic to create an anonymous closure at toplevel. Just write `let compare b1 b2 = ...`.